### PR TITLE
Stabilize 2018 edition

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
r? @eddyb 
cc @Centril @aturon

### CODE REVIEW

File: src/doc/rustdoc/src/command-line-arguments.md [hunk 1]
```diff
@@ -345,3 +345,17 @@ $ rustdoc src/lib.rs --sysroot /path/to/sysroot
 
 Similar to `rustc --sysroot`, this lets you change the sysroot `rustdoc` uses
 when compiling your code.
+
+### `--edition`: control the edition of docs and doctests
+
+Using this flag looks like this:
+
+```bash
+$ rustdoc src/lib.rs -Z unstable-options --edition 2018
```
Comment (Centril) [inline src/doc/rustdoc/src/command-line-arguments.md]: Hmm... it is stable but still using `unstable-options`?

**Comment by Mark-Simulacrum** [comment]
Okay, I believe this should work now, but still running tests locally.

An interesting thing I noticed is that `rust_2018_preview` is currently required to get some edition lints to fire, because otherwise the code won't report the warnings/errors. Is this expected? I'm not sure, but it feels like perhaps no. @alexcrichton, do you perhaps know?

If this is expected we might want to alias some other feature to it since once we ship I'd rather not have users writing "preview". cc @Centril, @aturon 

----

Otherwise I believe this is ready for review

r? @alexcrichton

**Comment by alexcrichton** [comment]
This looks good to me! r=me with green travis

And yeah it's expected that transition lints don't fire unless the unstable feature is enabled. Once everything stabilizes though they'll fire by default.

**Comment by Mark-Simulacrum** [comment]
Going to r+ this since I've run a good portion of tests locally...

@bors r=alexcrichton

**Comment by matthiaskrgr** [comment]
This is probably going to fail just as https://github.com/rust-lang/rust/pull/54036 due to crate vis modifier used as part 2018 without feature gate in clippy.

**Comment by Mark-Simulacrum** [comment]
Yes, that seems likely. I'll try to coordinate a fix. @bors r-

**Comment by matthiaskrgr** [comment]
https://github.com/rust-lang-nursery/rust-clippy/pull/3150 will make clippy compatible with this PR.

**Comment by matthiaskrgr** [comment]
Ah, then we'll also have to update clippy_lints used in RLS and push a RLS update to rustc ....

**Comment by mati865** [comment]
@matthiaskrgr `clippy_lints` version patched in this repository. So when you update Clippy submodule RLS will be build with that updated version.

**Comment by matthiaskrgr** [comment]
Ah, excellent, I didn't notice that :D

**Comment by matthiaskrgr** [comment]
You can see in the log that clippy build failed, yet bors didn't see this as a reason to mark the build as failed https://api.travis-ci.org/v3/job/425956240/log.txt

**Comment by Mark-Simulacrum** [comment]
I'll update this PR to include a clippy bump sometime today.

**Comment by matthiaskrgr** [comment]
In case you want it, I already have a rebased version with this PR + clippy bump in the pipeline but I didn't push yet because I wanted to wait until local testbuild is done.

**Comment by Mark-Simulacrum** [comment]
Yes, that would be great, thanks!

**Comment by matthiaskrgr** [comment]
Ok, it turned out that rls suffers from the same problem but we never got that far because clippy_lint dependency was already broken.
The PR fixing RLS is here: https://github.com/rust-lang-nursery/rls/pull/1050

I have to go AFK for most of the rest of the day in an hour or two, I allowed editing by maintainers on https://github.com/rust-lang/rust/pull/54057 so once the fix is merged into rls, you can hopefully add that to the PR and it should be good to go... :sweat_smile:

**Comment by Xanewok** [comment]
@Mark-Simulacrum @matthiaskrgr done, latest commit with the mentioned fix is https://github.com/rust-lang-nursery/rls/commit/5b5cd9d45719414196e254ec17baa598acc8cd25

**Pull request closed without merging**